src/config/prettier-config/load-external-config.js

```diff
@@ -15,7 +15,16 @@ async function loadExternalConfig(externalConfig, configFile) {
   3. is a dirname with index.js inside
   */
   try {
-    return requireFromFile(externalConfig, configFile);
+    const required = requireFromFile(externalConfig, configFile);
+    // Since Node.js v23 onwards, it is possible to load ESM using require.
+    // If that feature is enabled, it is necessary to return the default.
+    // https://github.com/prettier/prettier/issues/16812
+    // FIXME: We want to add tests but https://github.com/jestjs/jest/issues/15363 blocks
+    // @ts-expect-error
+    if (process.features.require_module && required.__esModule) {
+      return required.default;
+    }
+    return required;
   } catch (/** @type {any} */ error) {
```

Comment:
Maybe simply use `required?.default ?? required`?    https://github.com/babel/babel/issues/16441

Reply:
I prefer the current code. Because if the option had a `default` field, it would be used. I know that such a case rarely happens, but it's not that complicated code, so I prefer to be conservative.